ETL/Pipeline/extract.py

```diff
@@ -0,0 +1,61 @@
+"Extract Script: Pulls current price and sale status from ASOS API"
+from datetime import datetime
+import requests
+from lambda_multiprocessing import Pool
+
+
+def get_url(product_id: int) -> str:
+    """Returns the API URL for a given product on the ASOS website."""
+    if not isinstance(product_id, int):
+        raise TypeError('Product id must be a integer to get the url.')
+
+    return f"https://www.asos.com/api/product/catalogue/v4/stockprice?productIds=\
+        {product_id}&store=COM&currency=GBP&keyStoreDataversion=ornjx7v-36&country=GB"
+
+
+def get_product_info(product_data: int, headers: dict) -> int:
+    """Gets the price information for a specified product from the ASOS API."""
+    price_endpoint = get_url(product_data['product_id'])
```

Comment:
product_data is an integers but being treated as a dictionary. This needs clarifying

Reply:
Nice catch, this is a mistake. Its been corrected, it is a dict.